packages/core/test/features/st-var.spec.ts

```diff
@@ -0,0 +1,1219 @@
+                '/formatter.js': `
+                    module.exports = function fail() {
+                        throw new Error("FAIL!");
+                    }
+                `,
+                '/entry.st.css': `                
+                    @st-import fail from './formatter.js';
+
+                    :vars {
+                        definition: fail(input);
+                        color: red;
+                    }
+                    .root {
+                        /* @decl(definition) prop: fail(input) */
+                        prop: value(definition);
+
+                        /* @decl(value) prop: fail(a, red, z) */
+                        prop: fail(a, value(color), z);
+                    }
+                `,
+            });
+
+            const { meta, exports } = sheets['/entry.st.css'];
+
+            // symbols
+            expect(STVar.get(meta, `definition`), `definition symbol`).to.contain({
+                text: `fail(input)`,
+            });
+
+            // exports
+            expect(exports.stVars.definition, `definition JS export`).to.eql(`fail(input)`);
+        });
+    });
+    describe(`css-media`, () => {
+        it(`should resolve value() function`, () => {
+            const { sheets } = testStylableCore(`
+                :vars {
+                    mobile-width: 200px;
+                }
+
+                /* @atrule screen (200px) */
+                @media screen (value(mobile-width)) {}
+            `);
+
+            const { meta } = sheets[`/entry.st.css`];
+
+            shouldReportNoDiagnostics(meta);
+        });
+    });
+    describe(`hooks.replaceValueHook`, () => {
+        it(`should override value() result"`, () => {
+            let valueCallCount = 0;
+            const { sheets } = testStylableCore(
+                `
+                :vars {
+                    a: "red";
+                    b: green;
+                }
+                .container {
+                    /* @decl prop: override(0) red-a-true */
+                    prop: value(a);
+                    
+                    /* @decl prop: override(1) green-b-true */
+                    prop: value(b);
+                }
+            `,
+                {
+                    stylableConfig: {
+                        hooks: {
+                            replaceValueHook(resolved, name, isLocal) {
+                                return `override(${valueCallCount++}) ${resolved}-${name}-${isLocal}`;
+                            },
+                        },
+                    },
+                }
+            );
+
+            const { meta, exports } = sheets['/entry.st.css'];
+
+            shouldReportNoDiagnostics(meta);
+
+            // JS exports - ToDo: fix exports to use hook
+            expect(exports.stVars.a, `a JS export`).to.eql(`"red"`);
+            expect(exports.stVars.b, `b JS export`).to.eql(`green`);
+        });
+        it(`should override value() from JS import"`, () => {
```

Comment:
should move to value feature

Reply:
agreed, no value feature yet, so just a comment for now